pkg/agent/types.go

```diff
@@ -20,6 +20,7 @@ type KeyVaultRef struct {
 type NodeBootstrappingConfiguration struct {
 	ContainerService              *datamodel.ContainerService
 	CloudSpecConfig               *datamodel.AzureEnvironmentSpecConfig
+	K8sComponents                 map[string]string
```

Comment:
we only need a few value from this, hyperkube, pause,winzip and nodestatusfreq. Can we make a struct to hold them? I feel that's more explicit than a map

Reply:
Code updated to take in a struct with explicit listing of the strings we use.